src/gui/editors/PianoRoll.cpp

```diff
@@ -2114,31 +2107,23 @@ void PianoRoll::mouseMoveEvent( QMouseEvent * me )
 						m_pattern->instrumentTrack()->processInEvent( evt );
 					}
 				}
-				else
+				else if( n->isPlaying() )
 				{
-					if( n->isPlaying() )
-					{
-						// mouse not over this note, stop playing it.
-						m_pattern->instrumentTrack()->pianoModel()->handleKeyRelease( n->key() );
+					// mouse not over this note, stop playing it.
+					m_pattern->instrumentTrack()->pianoModel()->handleKeyRelease( n->key() );
 
-						n->setIsPlaying( false );
-					}
+					n->setIsPlaying( false );
 				}
 
-				// set textfloat visible if we're on a note
-				if( on_note )
-				{
-					s_textFloat->moveGlobal( this,	QPoint( me->x() + 4, me->y() + 16 ) );
-					s_textFloat->show();
-				}
-				else
-				{
-					s_textFloat->hide();
-				}
 
 				--it;
-
 			}
+
+			// Emit pattern has changed
+			m_pattern->dataChanged();
```

Comment:
This should not be a problem since `clearNotes` just emits this after deleting all notes, presuming it's correctly implemented.